### ISSUE TITLE
description missing when mixing inline + user-defined type

### ISSUE THREAD

**Issue by JMLX42** [juhaku/utoipa#797]
## How to reproduce

`utoipa` version 4.0.0

```rust
    #[schema(as = BufferResourceIdentifierObject)]
    pub struct BufferResourceIdentifierObject {
        #[serde(rename = "type")]
        #[schema(default = "buffers")]
        type_name: String,
        id: uuid::Uuid,
    }

    pub struct BufferResourceRelationshipToOne {
        #[schema(inline)]
        pub data: BufferResourceIdentifierObject,
    }

   #[serde(rename_all = "camelCase")]
    pub struct BufferViewResourceSparseRelationships {
        #[serde(default, skip_serializing_if = "Option::is_none")]
        #[schema(inline)]
        /// The index of the buffer.
        pub buffer: Option<crate::buffer::BufferResourceRelationshipToOne>,
    }
```

## Actual Result

For some reason, the `description` field is missing.

```json
{
  "relationships": {
    "type": "object",
    "properties": {
      "buffer": {
        "allOf": [
          {
            "type": "object",
            "required": [
              "data"
            ],
            "properties": {
              "data": {
                "type": "object",
                "required": [
                  "type",
                  "id"
                ],
                "properties": {
                  "type": {
                    "type": "string",
                    "default": "buffers"
                  },
                  "id": {
                    "type": "string",
                    "format": "uuid"
                  }
                }
              }
            }
          }
        ],
        "nullable": true
      }
    }
  }
}
```

## Expected Result

```json
{
  "relationships": {
    "type": "object",
    "properties": {
      "buffer": {
        "allOf": [
          {
            "description": "The index of the buffer.",
            "type": "object",
            "required": [
              "data"
            ],
```

**Comment by juhaku** [comment]
This probably hits a scenario that does not fully support all the fields. Thanks for reporting with actual example this will make the further investigation easier when begin.

**Comment by juhaku** [comment]
Fix is coming in PR #1102
